### PR TITLE
Remove static bucket hack

### PR DESCRIPTION
**Pull request by nitrocode**
This is for comments to ensure this PR doesn't break backwards compatibility.

## what
* This simplifies calculating the correct bucket regional domain name using the s3 bucket's `bucket_regional_domain_name` instead of the original hack from PR https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/pull/17
* I saw this output used on the official hashicorp docs for the [aws_cloudfront_distribution](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_distribution) resource

## why
* This simplifies the regional domain name
* Prevents having to maintain a list of buckets with dashes
* Prevents issues related to the data source of the `aws-cli` s3 bucket
* Should maintain destruction of the resources as it does before

## references
* More digging down the rabbit hole in this ticket https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/issues/101

### CODE REVIEW

**Review by aknysh — DISMISSED** [review]
@nitrocode 
Thanks, LGTM.
I think `var.use_regional_s3_endpoint` is not in use anymore and needs to be removed from 

https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/blob/master/variables.tf#L31

Also, please run 
```
make init
make github/init
make readme
```

and commit the changes.

Thanks

**Review by aknysh — APPROVED** [review]
thanks @nitrocode

File: main.tf [hunk 1]
```diff
@@ -190,13 +181,7 @@ locals {
       concat([var.origin_bucket], concat([""], aws_s3_bucket.origin.*.id))
     )
   )
-
-  bucket_domain_name = (var.use_regional_s3_endpoint || var.website_enabled) ? format(
-    var.website_enabled ? "%s.s3-website%s%s.amazonaws.com" : "%s.s3%s%s.amazonaws.com",
```
Comment (syphernl) [inline main.tf:195]: @nitrocode This change seems to have broken the "s3-website" functionality. It now wants to change the origin `domain_name` from 
`XXXXX.s3-website.eu-west-1.amazonaws.com` to `XXXXX.s3.eu-west-1.amazonaws.com` which results in an `AccessDenied` when loading the domain.

Comment (nitrocode) [reply main.tf:195]: Ah what we want is bucket regional domain name output of the bucket. I'll put in another pr

Comment (nitrocode) [reply main.tf:195]: I created this ticket https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/issues/148 after reviewing your comment. Good catch.

**Comment by alexjurkiewicz** [comment]
The removal of `use_regional_s3_endpoint` parameter was also a breaking change, and not communicated in release notes :(

**Pull request merged**